### ISSUE TITLE
Feature to view default TLS parameters such as TLS version, cipher used

### ISSUE THREAD

**Issue by vikasg603** [httptoolkit/httptoolkit#384]
Hey Team, We are using http toolkit from last 1 year and This is the best tool we can have, It does for what it is built.

Recently i saw 1 blog post regarding TLS [Fingerprinting](https://httptoolkit.com/blog/tls-fingerprinting-node-js/)

HTTP toolkit definitely randomise TLS fingerprint when sending a request but then it creates another problem, Problem when some TLS Cipher is in whitelist rather then in blocklist.

An example can be some android application where developer is using custom cipher and server whitelisted that cipher.

Currently I can't see any way to see what TLS parameters is used for the TLS handshaking between the client and the HTTP toolkit and not even between HTTP toolkit to the server.

I know this is bit tricky but even some workaround will work, Even a custom script in place of httptoolkit can solve my issue.

Thanks.

**Comment by pimterry** [comment]
Interesting, I haven't seen many examples of custom TLS configuration like you describe in the wild, but yes I can imagine how that could cause problems.

I'm open to adding this to HTTP Toolkit too, but right now it's not a top priority, and it's more complicated than it sounds! There's often not just one set of TLS parameters, there's a layer of proxy connection TLS + intercepted server TLS on the client side, plus upstream server TLS too, and potentially upstream proxy TLS, if one is configured. And some of these connections are shared by many requests: clients can make a single HTTP/2 proxy connection over TLS to HTTP Toolkit, and then open many individual separate HTTPS connections inside that tunnel to other upstream servers elsewhere.

It's all interesting, but it's hard to know how to easily represent all of that in the UI, and for most cases it's just noise. I am open to this though, if you have a good idea for the frontend design. If you want to help make this happen, I'd suggest starting some ideas, and opening a PR to [Mockttp](https://github.com/httptoolkit/mockttp/) to add the basic TLS data to the `request-initiated` event, at which point (after updating Mockttp everywhere) it'll become available to the UI. We do have quite a bit of that data available during processing via https://github.com/httptoolkit/read-tls-client-hello and Node's own TLS APIs, it's just not really exposed right now.

Alternatively, in the meantime, your quickest fix for this is probably [Wireshark](https://www.wireshark.org/). Wireshark can be complicated for some use cases, and can't do interactive interception (so it usually can't see inside encrypted connections and it can never rewrite traffic) _but_ it can view raw traffic, and the unencrypted raw TLS client & server hellos have all the data you're looking for I think. If you run that, you'll immediately see the outer TLS parameters used by HTTP Toolkit, on both the client and the server side. You might also find https://tls13.xargs.org/ a useful intro to interpreting that, if you haven't investigated this much in the past.

**Comment by ibrah3m** [comment]
+1 I 2nd this And I wish it gets implemented on httptoolkit.

Also Are we able to use Wireshark and httptoolkit at the same time ?
or linking the the httptoolkit android proxy to route the traffic to Wireshark @pimterry 
thanks 👋,

**Comment by pimterry** [comment]
> Also Are we able to use Wireshark and httptoolkit at the same time ?

Yes - HTTP Toolkit and Wireshark appear similar but internally function completely differently, and so both of them can be used at the same time. If you want to intercept traffic with HTTP Toolkit and inspect lower level non-HTTP data too (raw IP & TCP, or TLS server & client handshake data) then you just need to set up HTTP interception with HTTP Toolkit as normal, and then use Wireshark on the same computer to monitor packets so you can dig into the raw data beneath.

To filter all the data that appears in Wireshark so you can see the client's connection details easily, you can filter for incoming traffic to HTTP Toolkit's proxy port. To filter to see just the upstream traffic details you'll need to filter for the specific upstream addresses that you're interested in.

**Comment by MrNewbie4Ever** [comment]
+1 this could help a lot to see how TLS and what type of TLS request was using, like mozilla has.
